api/rest/client_test.go

```diff
@@ -328,6 +329,30 @@ func TestClient_WithRetries(t *testing.T) {
 	assert.Equal(t, 2, apiHits)
 }
 
+func TestClient_WithCustomRetriesOnRequest(t *testing.T) {
+	apiHits := 0
+	server := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		rw.WriteHeader(http.StatusTooManyRequests)
+		apiHits++
+	}))
+	defer server.Close()
+
+	baseURL, _ := url.Parse(server.URL)
+	client := NewClient(baseURL, nil, WithRetryOptions(&RetryOptions{
+		MaxRetries: 10,
+		ShouldRetryFunc: func(resp *http.Response) bool {
+			return resp.StatusCode != http.StatusOK
+		},
+	}))
+
+	ctx := testutils.ContextWithLogger(t)
+
+	resp, err := client.GET(ctx, "", RequestOptions{MaxRetries: testutils.Pointer(1)})
+	require.NoError(t, err)
+	assert.Equal(t, http.StatusTooManyRequests, resp.StatusCode)
+	assert.Equal(t, 2, apiHits)
+}
```

Comment:
Shall we add a test with a custom `DelayAfterRequest`?

Reply:
The thing is, how would you test that? I could test the merge function instead. This one is private, but in the same package

Reply:
I was curious about how to test it, so I added one anyway 😅